.github/workflows/upstream-dev-ci.yaml

```diff
@@ -11,10 +11,39 @@ on:
   workflow_dispatch: # allows you to trigger the workflow run manually
 
 jobs:
+  detect-ci-trigger:
+    name: detect upstream-dev ci trigger
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    outputs:
+      triggered: ${{ steps.detect_trigger.outputs.CI_TRIGGERED }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+         fetch-depth: 5
+         ref: ${{ github.ref }}
```

Comment:
@keewis, here is the workaround I come up with. Since I don't know how to get rid of the merge commits created by `actions/checkout`, I just increased the value for `-n` in `git log -n` by 1. 

Let me know if this achieves what you are going for...

Reply:
thanks for investigating, @andersy005. Unfortunately this only works for `pull_request` but not `push`. I added another step to change the used ref depending on `github.event_name`. Hopefully it now works correctly.

Since we probably have to do the same for `skip-ci` in #4730, we could move all of this into a custom action in a new PR.

Reply:
> Since we probably have to do the same for skip-ci in #4730, we could move all of this into a custom action in a new PR.

👍🏽